### PR TITLE
fix clippy warnings for tools and run lints for pull requests

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**

#1776 (partial fix)

**Description of changes:**
Fix or silence all clippy warnings for `tools`, and add a step to the pull request workflow so they stay fixed.

**Testing done:**
Verified that builds completed via `buildsys`.

Ran the following `pubsys` related tasks:
* ami
* repo
* validate-repo
* check-repo-expirations
* refresh-repo
* ssm
* promote-ssm
* upload-ova
* vmware-template

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by bcressey** [comment]
Let me run `rustup` and fix some more lints. 😆

File: Makefile.toml [hunk 1]
```diff
@@ -366,6 +366,25 @@ fi
...
 ]
 
+[tasks.check-lints]
+script = [
+'''
+rc=0
+
+# For rust first-party source code
+if ! cargo clippy \
+  --manifest-path ${BUILDSYS_TOOLS_DIR}/Cargo.toml \
```
Comment (etungsten) [inline Makefile.toml:376]: I guess the comment isn't exactly accurate yet. Are we planning to do this to first-party code in `/sources` as a follow up?

**Review by etungsten — APPROVED** [review]

File: Makefile.toml (hunk 1 above)
Comment (bcressey) [reply Makefile.toml:376]: Yes, I'd like first-party code to be covered as well, but it's more than I had the time to tackle today. I was working on an unrelated update to buildsys and got annoyed by the existing `clippy` warnings. 😀

**Review by zmrow — APPROVED** [review]
:safety_vest:

**Review by cbgbt — APPROVED** [review]
:rocket:

**Pull request merged**